src/extension.ts

```diff
@@ -170,9 +171,17 @@ async function startClients(folders = prunedFolderNames()) {
 		if (!clients.has(folder)) {
 			const commandPrefix = shortUid.seq();
 			const client = newClient(command, folder, commandPrefix);
+			client.onReady().then(() => {
+				reporter.sendTelemetryEvent('startClient');
+			});
+			client.onDidChangeState((event) => {
+				if (event.newState === ClientState.Stopped) {
+					clients.delete(folder);
```

Comment:
Is there a reason `clients.delete` should be called from here, as opposed to `stopClients`?

Reply:
This way we can ensure it's always in sync with when the client actually stops, even if that's triggered by something other than our own code.

Reply:
I see! 👍🏻